bridged-nft-token/src/nft_core.rs

```diff
@@ -0,0 +1,289 @@
+use crate::*;
+use near_sdk::json_types::{ValidAccountId, U64};
+use near_sdk::{ext_contract, log, Gas, PromiseResult};
+
+const GAS_FOR_NFT_APPROVE: Gas = 10_000_000_000_000;
+const GAS_FOR_RESOLVE_TRANSFER: Gas = 10_000_000_000_000;
+const GAS_FOR_NFT_TRANSFER_CALL: Gas = 25_000_000_000_000 + GAS_FOR_RESOLVE_TRANSFER;
+const NO_DEPOSIT: Balance = 0;
+
+pub trait NonFungibleTokenCore {
+    fn nft_transfer(
+        &mut self,
+        receiver_id: ValidAccountId,
+        token_id: TokenId,
+        approval_id: Option<U64>,
+        memo: Option<String>,
+    );
+
+    /// Returns `true` if the token was transferred from the sender's account.
+    fn nft_transfer_call(
+        &mut self,
+        receiver_id: ValidAccountId,
+        token_id: TokenId,
+        approval_id: Option<U64>,
+        memo: Option<String>,
+        msg: String,
+    ) -> Promise;
+
+    fn nft_approve(&mut self, token_id: TokenId, account_id: ValidAccountId, msg: Option<String>);
+
+    fn nft_revoke(&mut self, token_id: TokenId, account_id: ValidAccountId);
+
+    fn nft_revoke_all(&mut self, token_id: TokenId);
+
+    fn nft_total_supply(&self) -> U64;
+
+    fn nft_token(&self, token_id: TokenId) -> Option<JsonToken>;
+}
+
+#[ext_contract(ext_non_fungible_token_receiver)]
+trait NonFungibleTokenReceiver {
+    /// Returns `true` if the token should be returned back to the sender.
+    fn nft_on_transfer(
+        &mut self,
+        sender_id: AccountId,
+        previous_owner_id: AccountId,
+        token_id: TokenId,
+        msg: String,
+    ) -> Promise;
```

Comment:
`nft_on_transfer` doesn't return `PromiseOrValue<bool>` as we have in the canonical example.

Spec:
https://nomicon.io/Standards/NonFungibleToken/Core.html#receiver-interface

There are a few instances of this in the `near-sdk-rs` canonical examples, but here's one:
https://github.com/near/near-sdk-rs/blob/de81ae5e21a406219911527ae4a622b1b3a2f633/near-contract-standards/src/non_fungible_token/core/core_impl.rs#L34-L43

There's also a receiver contract in there if that's more helpful

Reply:
it was removed and replaced by the one contained inside the standard